cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -925,13 +930,119 @@ def make_resource(self, data, **kwargs):
         return Timeouts(**data)
 
 
+class BudgetNotificationSchema(BaseSchema):
+    """Represent the schema of each notification under the NotificationsWithSubscribers field of a budget."""
+
+    notification_type = fields.Str(
+        metadata={"update_policy": UpdatePolicy.SUPPORTED},
+        validate=validate.OneOf(["ACTUAL", "FORECASTED"]),
+    )
+    comparison_operator = fields.Str(
+        metadata={"update_policy": UpdatePolicy.SUPPORTED},
+        validate=validate.OneOf(["GREATER_THAN", "EQUAL_TO", "LESS_THAN"]),
+    )
+    threshold = fields.Int(required=True, metadata={"update_policy": UpdatePolicy.SUPPORTED})
+    threshold_type = fields.Str(
+        metadata={"update_policy": UpdatePolicy.SUPPORTED},
+        validate=validate.OneOf(["ABSOLUTE_VALUE", "PERCENTAGE"]),
+    )
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return BudgetNotification(**data)
+
+
+class BudgetSubscriberSchema(BaseSchema):
+    """Represent the schema of an individual subscriber of a budget notification."""
+
+    subscription_type = fields.Str(
+        metadata={"update_policy": UpdatePolicy.SUPPORTED},
+        validate=validate.OneOf(["EMAIL", "SNS"]),
+    )
+    address = fields.Str(required=True, metadata={"update_policy": UpdatePolicy.SUPPORTED})
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return BudgetSubscriber(**data)
+
+
+class BudgetNotificationWithSubscribersSchema(BaseSchema):
+    """Represent the schema of the NotificationsWithSubscriber field of a budget."""
+
+    notification = fields.Nested(
+        BudgetNotificationSchema,
+        equired=True,
+        metadata={"update_policy": UpdatePolicy.SUPPORTED},
+    )
+
+    subscribers = fields.Nested(
+        BudgetSubscriberSchema,
+        required=True,
+        many=True,
+        metadata={"update_policy": UpdatePolicy.SUPPORTED, "update_key": "Name"},
+    )
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return BudgetNotificationWithSubscribers(**data)
+
+
+class BudgetLimitSchema(BaseSchema):
+    """Represent the schema of the BudgetLimit field of a budget."""
+
+    amount = fields.Int(required=True, metadata={"update_policy": UpdatePolicy.SUPPORTED})
```

Comment:
Is this amount require to be integer? What about float?

Reply:
Good call, it's actually a float. Updated!